lib/tasks/scheduler.rake

```diff
@@ -40,29 +40,65 @@ task :collect_zpids => :environment do
   puts "Missing ZPIDs for #{missing_zpids.reload.where(zpid: nil).count}"
 end
 
+NeighborhoodCsvHeaders = %w(
+  neighborhood_name
+  neighborhood_zip_code
+  house_street_address
+  house_city
+  house_state
+  house_zip_code
+  house_bedrooms
+  house_bathrooms
+  house_square_feet
+).freeze
+
+desc "Generate a csv to onboard a neighborhood"
+task generate_hood_onboarding_csv: :environment do |t|
+  require "csv"
+
+  filename = "#{DateTime.now.to_i}_onboard_neighborhood.csv"
+  CSV.open(filename, "w") { |csv| csv << NeighborhoodCsvHeaders }
+
+  Rails.logger.info "Created #{filename}"
+end
+
 desc 'Upload a CSV of a neighborhood'
-task :upload_neighborhood, [:neighborhood_name, :neighborhood_zip_code, :neighborhood_data] => :environment do |t, args|
+task :upload_neighborhood, [:hood_onboarding_csv_path] => :environment do |t, args|
   starting_count = House.count
 
-  Rails.logger.info <<~MSG.strip
-    There are #{starting_count} houses in the database.
-    You have passed in #{args.count} arguments, with these keys: #{args.keys}
-  MSG
+  csv_entries = CSV.read(args[:hood_onboarding_csv_path], headers: true).map(&:to_h)
+
+  if csv_entries.flat_map(&:keys).uniq != NeighborhoodCsvHeaders
+    error = <<~ERR
+      #{args[:hood_onboarding_csv_path]} does not have valid headers.
+      Got: #{csv_entries.flat_map(&:keys).uniq}
+      Want: #{NeighborhoodCsvHeaders}
+    ERR
+
+    raise ArgumentError, error
+  end
+
+  csv_entries.each do |csv_entry|
+    hood = Hood.find_or_create_by(name: csv_entry["neighborhood_name"], zip_code: csv_entry["neighborhood_zip_code"])
 
-  hood = Hood.find_or_create_by(name: args[:neighborhood_name], zip_code: args[:neighborhood_zip_code])
+    street_address = csv_entry["house_street_address"]&.strip
+    city           = csv_entry["house_city"]&.strip
+    state          = csv_entry["house_state"]&.strip
+    zip_code       = csv_entry["house_zip_code"]&.strip
+    bedrooms       = csv_entry["house_bedrooms"]&.strip
+    bathrooms      = csv_entry["house_bathrooms"]&.strip
+    square_feet    = csv_entry["house_square_feet"]&.strip
 
-  args[:neighborhood_data].each do |row|
-    street_address = row[0]&.strip
-    city           = row[1]&.strip
-    state          = row[2]&.strip
-    zip_code       = row[3]&.strip
-    bedrooms       = row[4]&.strip
-    bathrooms      = row[5]&.strip
-    square_feet    = row[6]&.strip
+    if street_address && city && state && zip_code
+      house = hood.houses.create(address: {
+        street_address: street_address,
+        city: city,
+        state: state,
+        zip_code: zip_code
+      })
 
-    address = {street_address: street_address, city: city, state: state, zip_code: zip_code}
-    h = House.create(address: address, hood: hood)
-    HouseMetadatum.create(bedrooms: bedrooms, bathrooms: bathrooms, square_feet: square_feet, house: h)
+      HouseMetadatum.create(bedrooms: bedrooms, bathrooms: bathrooms, square_feet: square_feet, house: house)
+    end
```

Comment:
Can we kick off the `collect_zpids` task at the end of this one so it fetches all the zpids once the houses have been uploaded?

Or you think we should keep them separate?

Reply:
Maybe should be a separate thing - it looks like this task is getting big

Reply:
Separate for now, IMO.